dowhy/causal_estimator.py

```diff
@@ -104,57 +86,68 @@ def __init__(
         :returns: an instance of the estimator class.
         """
-        self._data = data
         self._target_estimand = identified_estimand
-        # Currently estimation methods only support univariate treatment and outcome
-        self._treatment_name = treatment
-        self._outcome_name = outcome[0]  # assuming one-dimensional outcome
-        self._control_value = control_value
-        self._treatment_value = treatment_value
+
         self._significance_test = test_significance
         self._effect_strength_eval = evaluate_effect_strength
-        self._target_units = target_units
-        self._effect_modifier_names = effect_modifiers
         self._confidence_intervals = confidence_intervals
-        self._bootstrap_estimates = None  # for confidence intervals and std error
-        self._bootstrap_null_estimates = None  # for significance test
-        self._effect_modifiers = None
+
         self.method_params = kwargs
         # Setting the default interpret method
         self.interpret_method = CausalEstimator.DEFAULT_INTERPRET_METHOD
 
         self.logger = logging.getLogger(__name__)
 
-        # Setting treatment and outcome values
-        if self._data is not None:
-            self._treatment = self._data[self._treatment_name]
-            self._outcome = self._data[self._outcome_name]
-
-        if self._effect_modifier_names:
-            # only add the observed nodes
-            self._effect_modifier_names = [
-                cname for cname in self._effect_modifier_names if cname in self._data.columns
-            ]
-            if len(self._effect_modifier_names) > 0:
-                self._effect_modifiers = self._data[self._effect_modifier_names]
-                self._effect_modifiers = pd.get_dummies(self._effect_modifiers, drop_first=True)
-                self.logger.debug("Effect modifiers: " + ",".join(self._effect_modifier_names))
-            else:
-                self._effect_modifier_names = None
-
         # Check if some parameters were set, otherwise set to default values
         self.num_null_simulations = num_null_simulations
         self.num_simulations = num_simulations
         self.sample_size_fraction = sample_size_fraction
         self.confidence_level = confidence_level
         self.num_quantiles_to_discretize_cont_cols = num_quantiles_to_discretize_cont_cols
         # Estimate conditional estimates by default
+        self.need_conditional_estimates = need_conditional_estimates
+
+        self._bootstrap_estimates = None
+        self._bootstrap_null_estimates = None
+
+    def set_data(self, data: pd.DataFrame, treatment_name: List[str], outcome_name: List[str]):
+        """Sets the data for the estimator
+        :param data: data frame containing the data
+        :param treatment: name of the treatment variable
+        :param outcome: name of the outcome variable
+        """
+        # Currently estimation methods only support univariate treatment and outcome
```

Comment:
Oh, I copied this comment from: https://github.com/py-why/dowhy/blob/main/dowhy/causal_estimator.py#L109 I guess that it is a legacy comment that was never removed? 

If that's the case I'll need to update the types for all treatment parameters and the docs, because from the text of it (at least to me) it says that it is a single string and not a list of strings.

Reply:
oh I see. yeah, that's my bad. I may have updated the code without the docstring. 

But yes, we do support multiple treatments for some methods, that's why treatment is always passed as a list.